### DISCUSSION TITLE
ubereats.com: ads

### DISCUSSION THREAD

**Discussion by aldobarr** [uBlockOrigin/uAssets#23798] [General] [answered]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL(s) where the issue occurs.

```yaml
https://www.ubereats.com/orders/{ORDER-ID}
```

### Description

There are some new ads on the order page after an order has been placed.

### Other extensions used

betterTTV
bitwarden
clickbait remover for youtube
don't fuck with paste
greasemonkey (only 1 script which only interacts on twitch.tv)
hide private mode
return youtube dislike
sponsorblock for youtube
ttv lol pro
translate web pages
youtube high definition
youtube nonstop
youtube spam remover

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/5aaeca6a-725f-4574-8121-cbb8be5c5ca9)

</details>

### Configuration

<details>

```yml
uBlock Origin: 1.57.2
Firefox: 125
filterset (summary):
 network: 187584
 cosmetic: 49816
 scriptlet: 24739
 html: 2061
listset (total-discarded, last-updated):
 added:
  https://gitlab.com/magnolia1234/bypass-paywalls-clean-filters/-/raw/main/bpc-paywall-filter.txt: 0-0, never
  https://raw.githubusercontent.com/liamengland1/miscfilters/master/antipaywall.txt: 2098-0, 5m
  adguard-spyware-url: 1512-120, 8m
  fanboy-thirdparty_social: 68-0, 8m
  ublock-annoyances: 5823-38, 8m
  adguard-spyware: 83498-31917, 8m
 default:
  user-filters: 5-0, never
  ublock-filters: 38196-118, 9m
  ublock-badware: 8548-0, 9m
  ublock-privacy: 907-3, 9m
  ublock-unbreak: 2332-2, 8m Δ
  ublock-quick-fixes: 193-5, 8m Δ
  easylist: 87514-169, 8m
  easyprivacy: 50907-75, 8m
  urlhaus-1: 12635-3, 8m
  plowe-0: 3729-1160, 8m
filterset (user): [array of 5 redacted]
trustedset:
 added: [array of 7 redacted]
hostRuleset:
 added: [array of 8 redacted]
urlRuleset:
 added: [array of 442 redacted]
userSettings:
 advancedUserEnabled: true
hiddenSettings: [none]
supportStats:
 allReadyAfter: 154 ms (selfie)
 maxAssetCacheWait: 43 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 74
 network:
  ubereats.com: 74
 extended:
  ##.dw.ec
  ##+js(set-constant, Navigator.prototype.globalPrivacyControl, fa…
  ##+js(set-constant, navigator.globalPrivacyControl, false)
```

</details>

**Comment by aldobarr** [comment]
@stephenhawk8054 I order quite often. Happy to help with my account if needed.

**Comment by stephenhawk8054** [comment]
@aldobarr The label is just to inform any others who want to look at the case that the issue needs an account to reproduce.

Can you right click on the ads -> Click on `Inspect` -> It will open Firefox devtools -> Screenshot it here? (It would be better if you can drag the devtools to be bigger before screenshot)

**Comment by aldobarr** [comment]
> @aldobarr The label is just to inform any others who want to look at the case that the issue needs an account to reproduce.
> 
> Can you right click on the ads -> Click on `Inspect` -> It will open Firefox devtools -> Screenshot it here? (It would be better if you can drag the devtools to be bigger before screenshot)

@stephenhawk8054 The last time I ordered I didn't get an ad. Just a self promo to upgrade ubear eats pro. Example:

![no_ad](https://github.com/uBlockOrigin/uAssets/assets/1071197/8c1d8cac-aa27-4555-9ed2-d5432fbcec50)

Thought you might want to see that too. I got the ads back this morning:
![yes_ad](https://github.com/uBlockOrigin/uAssets/assets/1071197/cc626660-d86a-4da7-948f-034a513e24b9)

Included the actual dom contents as a text file as well in both cases.

**Comment by stephenhawk8054** [comment]
Yeah, this is difficult to make if just based on the DOM. You can try this as a custom filter

```adb
ubereats.com##:matches-path(/orders) main#main-content ol[style^="transform"]:has(> li > a[href]):has-text(Ad)
```

but I'm not sure if it would cause breakages somewhere else or not.

**Comment by aldobarr** [comment]
Order just got delivered so I'll try it out next time. Also realistically the only thing that matters on the order page is the delivery estimate and the map. The rest is just fluff imo.

**Comment by stephenhawk8054** [comment]
I'll convert this to a discussion. It's still a risky filter so anyone can test it as a custom filter

**Comment by aldobarr** [comment]
> Yeah, this is difficult to make if just based on the DOM. You can try this as a custom filter
> 
> ```
> ubereats.com##:matches-path(/orders) main#main-content ol[style^="transform"]:has(> li > a[href]):has-text(Ad)
> ```
> 
> but I'm not sure if it would cause breakages somewhere else or not.

@stephenhawk8054 before:
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/19a26260-167d-4901-8462-8490c954e9c9)

After:
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/055f6eab-7e11-4bd5-b03e-3f8ff8d0d9fc)


Looks good to me

**Comment by aldobarr** [comment]
The ad is back so the previous filter isn't catching it anymore

**Reply by Yuki2718** [reply]
As it requires an account, sharing screenshots of DOM as in https://github.com/uBlockOrigin/uAssets/discussions/23798#discussioncomment-9522325 will be appreciated.

**Reply by aldobarr** [reply]
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/1dc866e5-ce31-48c6-8f3c-755ee9a263ad)
[ubereats.txt](https://github.com/user-attachments/files/16071772/ubereats.txt)

**Reply by Yuki2718** [reply]
@aldobarr Can you also take a screenshot of the top tag of the ad? I mean climbing up the DOM tree to the topmost position that still only selects the ad. I'm interested in CSS property of the element (the right-side column).

Meanwhile, test if
```
ubereats.com##:matches-path(/orders) #main-content div:not([class]) div[class*=" "] > a[class*=" "] > div[class*=" "] > div[class*=" "] > div[class]:has(> div[class*=" "] > div[class*=" "] > div[class*=" "]:has-text(/^Ad$/)):style(border: 16px dotted pink !important;)
```

properly highlights the ad with a dotted line.

**Reply by jaredcat** [reply]
@Yuki2718 

<img width="542" alt="Screenshot 2024-07-03 at 1 13 33 PM" src="https://github.com/uBlockOrigin/uAssets/assets/3268576/f41b8d7f-3788-462f-bc3a-2a4bb92e252b">
<img width="1147" alt="Screenshot 2024-07-03 at 1 14 41 PM" src="https://github.com/uBlockOrigin/uAssets/assets/3268576/d4cac888-297c-4018-9fdb-051207626487">

<img width="1206" alt="Screenshot 2024-07-03 at 1 18 21 PM" src="https://github.com/uBlockOrigin/uAssets/assets/3268576/20b75069-10e8-473f-b9b8-890d208b55dc">


```html
<div class="h5 gc"><div><div class="bu bh"><div class="h6 d8 al cm"><div class="bh al ak fq cm"><div class="al f6 fr fs"><a href="https://redeem.services.apple/uberappletv?itscg=MC_30000&amp;itsct=atvp_brand_omd&amp;mttn3pid=uber&amp;mttnagencyid=a5e&amp;mttncc=US&amp;mttnsiteid=143238&amp;mttnsubad=04343586" class="ej bh bu af h7 fw ci h8 h9"><div class="ae al am cn"><div class="ak bh ha hb hc hd"><img alt="" role="presentation" src="https://d1g1f25tn8m2e6.cloudfront.net/3781a5a8-3194-44fd-931e-d1cbe9ec0d4d.png" class="he bh ae ag as"></div><div class="fv g0 ak al am hf et"><div class="bh"><div class="b1 bk h2 bm h3"><div class="hg hh hi bu">You’ve unlocked a 30 day free trial of Apple TV+.</div></div><div class="b1 hj bo fd bq e2"><div class="hg hh hi bu">Watch Joel Edgerton &amp; Jennifer Connelly</div></div></div><div class="bh" uj6r4fap9="" ds56w4nil=""><div class="al et bc hk"><div class="al cm hl"><div class="ck hm hn ho hl gv af hp bo hq e1 gq"><div class="hg hh hr bu">Redeem now</div>→</div></div><div class="b1 bo hq e1 gq"><div class="hg hh hr bu">Ad</div></div></div></div></div></div></a></div></div></div></div></div></div>
```

**Reply by jaredcat** [reply]
I was able to get to this (just one div off) before my food was delivered:

<img width="1068" alt="Screenshot 2024-07-03 at 1 27 01 PM" src="https://github.com/uBlockOrigin/uAssets/assets/3268576/701f4c38-da67-47d7-a209-2944bb642471">

with: 
```
ubereats.com##:matches-path(/orders) #main-content div:not([class]):has(div[class*=" "] > a[class*=" "] > div[class*=" "] > div[class*=" "] > div[class] > div[class*=" "] > div[class*=" "] > div[class*=" "]:has-text(/^Ad$/)):style(border: 16px dotted pink !important;)
```

**Comment by aldobarr** [comment]
The ad was hidden today so whatever update was added for it looks good:
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/990fb271-e1d0-421c-860a-296d57ed1fbd)


My only complaint would be that the parent element gets left in as undesirable empty space:
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/03c5993c-ffed-41ea-935c-40196b95d03e)

But it's a very minor complaint. This is already good as no ad with white space is better than ad 1 trillion %.

**Reply by Yuki2718** [reply]
We didn't update anything.

**Reply by stephenhawk8054** [reply]
The ad does not appear simply because their link matches with a generic filter in EasyList

```adb
##a[href^="https://ad.doubleclick.net/"]
```

**Reply by aldobarr** [reply]
That's weird, I definitely didn't change anything and it wasn't being caught before

**Reply by stephenhawk8054** [reply]
The previous links did not start with `https://ad.doubleclick.net/`, that's why they were not caught.

**Comment by stephenhawk8054** [comment]
It might still leave placeholders and could be a whack-a-mole but we can add the filters based on the ad link if you can right click on it and paste the links when reporting to here whenever you see the ads. The filters could be safer for false positives as well.

For example, if the ad link is `https://redeem.services.apple.com/`, the filters could look like this:

```adb
ubereats.com##:matches-path(/orders) main#main-content div:not([class]) a[class][href^="https://redeem.services.apple.com/"]
```

**Reply by aldobarr** [reply]
I actually think this page doesn't need to be so carefully targeted really. The only information of value ever displayed on an order's page is the map for tracking where your driver is and maybe the delivery estimate. The rest is honestly worthless.

**Reply by stephenhawk8054** [reply]
If a filter is added to the main list, we need to minimize the chance of false positives, especially for the pages we cannot access. To you it might not be an important page, but we cannot speak on behalf of other users and they will submit issues to complain if breakage happens. The filter lists are not designed for just few users.

**Reply by Yuki2718** [reply]
I also guess the word "Ad" is dependent on your locale settings.

**Comment by aldobarr** [comment]
@Yuki2718 Here's another one with the top most element selected
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/40db843f-6b7b-4c17-a53c-3e9c6ca7a569)

**Reply by Yuki2718** [reply]
https://github.com/uBlockOrigin/uAssets/discussions/23798#discussioncomment-9952263 should still match. If you wanna hide, just remove `:style(border: 16px dotted pink !important;)`.

**Reply by jaredcat** [reply]
@Yuki2718 but, can you modify it so it selects the top most div? Currently, it'll leave that empty div, very similar looking to https://github.com/uBlockOrigin/uAssets/discussions/23798#discussioncomment-9961271

**Reply by Yuki2718** [reply]
Ah, didn't know it leaves placeholder. Here you go `ubereats.com##:matches-path(/orders) #main-content > div > div > div[class*=" "]:has(> div:not([class]) div[class*=" "] > a[class*=" "] > div[class*=" "] > div[class*=" "] > div[class] > div[class*=" "] > div[class*=" "] > div[class*=" "]:has-text(/^Ad$/)):style(border: 16px dotted pink !important;)` It looks the CSS property of the top element is not very useful by itself, at best we may use it to make the rule a bit safer.

**Reply by aldobarr** [reply]
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/9734b5d6-b4bd-4694-a738-9fc08b785ec7)

This filter looks good. Nothing else on the page is selected.

without style:
![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/17655f40-16cb-42a0-90e3-5f8e4222d82e)

No white space :D

**Reply by Yuki2718** [reply]
Can someone change your language settings in ubereats and see if the rule is still applied? Probably not, and in that case it will be nice to gather the keyword for each languages.

**Reply by aldobarr** [reply]
I changed the locale to es-ES and nothing changed on uber's side so I don't think it matters. But the ad did come back either way:

![firefox_2024-07-11_13-34-06](https://github.com/uBlockOrigin/uAssets/assets/1071197/d07b370d-acfa-49d8-bea3-e01e444e2143)

**Reply by aldobarr** [reply]
@Yuki2718 The previous filter worked this morning so I think it alternates.

**Reply by Yuki2718** [reply]
Test `ubereats.com##:matches-path(/orders) #main-content > div > div > div[class*=" "]:has(> div:not([class]) [class*=" "] > a[class*=" "] > div[class*=" "] > div[class*=" "] > div[class] > div[class*=" "] > div[class*=" "] > div[class*=" "]:has-text(/^Ad$/)):style(border: 16px dotted pink !important;)`

**Reply by Yuki2718** [reply]
It seems `.bh` is stable, maybe we can use this to make the rule safer.  
```
ubereats.com##:matches-path(/orders) #main-content > div > div > div[class*=" "]:has(> div:not([class]) [class*=" "] > a[class*=" "] > div[class*=" "] > div[class*=" "] > div.bh > div[class*=" "] > div[class*=" "] > div[class*=" "]:has-text(/^Ad$/)):style(border: 16px dotted pink !important;)
```

**Reply by aldobarr** [reply]
Do you want me to remove the previous filter? Also just to clarify if you want me to test the first commented filter, the second one or both? (Also, not planning on ordering anything else today so I'll try to get back to you tomorrow)

**Reply by Yuki2718** [reply]
The first one should cover both the new ad and old, so the previous rule can be removed. The second one is slightly mor targetted, if this works it's better. But it might stop working in future.

**Reply by aldobarr** [reply]
@Yuki2718 Here's the first one:
![image](https://github.com/user-attachments/assets/7831b772-a8f5-4c35-8af3-e35c29125ae8)

and here's the second:
![image](https://github.com/user-attachments/assets/a8420963-6ce3-45e9-aa4e-a4f8684140dd)

Both seem to work and leave no extra whitespace, at least for the ad I got today. I'll let you know if either stop working.

**Comment by aldobarr** [comment]
@Yuki2718 The last filter for the ads on the left side are still working, but I'm now getting these popup(modal) ads:
![image](https://github.com/user-attachments/assets/bf543c84-8260-45a2-b930-99aafc5aa368)

**Reply by Yuki2718** [reply, marked as answer]
I think this should be addressed in Annoyances filters. Does `ubereats.com##div[data-focus-lock-disabled="false"]:has(> #eater-message-desktop-interstitial-v2 > #eater-message-desktop-interstitial-modal)` work for you?

**Reply by aldobarr** [reply]
I do have annoyances enabled but I get that popup anyway. Shows up on the homepage too, so doesn't have to be after ordering.

I'll try that filter and let you know if I see it again

**Reply by aldobarr** [reply]
> I think this should be addressed in Annoyances filters. Does `ubereats.com##div[data-focus-lock-disabled="false"]:has(> #eater-message-desktop-interstitial-v2 > #eater-message-desktop-interstitial-modal)` work for you?

I haven't experienced this popup on either the home page or the order page since adding this filter 3 weeks ago. If this is part of annoyances though, not sure why I was getting it :/

**Reply by Yuki2718** [reply]
Currently no annoyances filter lists has that filter included. If the rule is safe enough, I will add this to AdGuard and Fanboy Annoyances.

**Reply by aldobarr** [reply]
> Currently no annoyances filter lists has that filter included. If the rule is safe enough, I will add this to AdGuard and Fanboy Annoyances.

I see, I haven't noticed any adverse effects. It's working fine for me
